frame/staking/src/lib.rs

```diff
@@ -1551,31 +1718,33 @@ impl<T: Trait> Module<T> {
 					total,
 				};
 
-				if exposure.total < slot_stake {
-					slot_stake = exposure.total;
-				}
-				<Stakers<T>>::insert(&c, exposure.clone());
+				total_staked = total_staked.saturating_add(exposure.total);
+				<ErasStakers<T>>::insert(&current_era, &c, exposure.clone());
 			}
 
-			// Update slot stake.
-			<SlotStake<T>>::put(&slot_stake);
-
-			// Set the new validator set in sessions.
-			<CurrentElected<T>>::put(&elected_stashes);
+			// Insert current era informations
+			<ErasTotalStake<T>>::insert(&current_era, total_staked);
+			let default_pref = ValidatorPrefs::default();
+			for stash in &elected_stashes {
+				let pref = all_validators_and_prefs.get(stash)
+					// This should always succeed but better to be safe
+					.unwrap_or(&default_pref);
+				<ErasValidatorPrefs<T>>::insert(&current_era, stash, pref);
```

Comment:
Here we can write the ErasValidatorPrefs for the current_era + 1 so nominators can nominate while knowing the preference in advance.